### DISCUSSION TITLE
Debug on MacOS

### DISCUSSION THREAD

**Discussion by xq114** [xmake-io/xmake-vscode#138] [Q&A]
https://github.com/xmake-io/xmake-vscode/blob/HEAD/src/debugger.ts#L106

这里单单对mac设置了externalconsole为true；这是为什么？ https://github.com/microsoft/vscode-cpptools/issues/5079 根据cpptools，mac上设置externalconsole是有问题的。

https://github.com/xmake-io/xmake-vscode/blob/HEAD/src/debugger.ts#L66

这里提到使用codelldb，但是在ReadMe里面对此没有任何说明；可否出一个教程说明一下使用codelldb进行debug的方法？

**Comment by waruqi** [comment]
> 这里单单对mac设置了externalconsole为true；这是为什么？

最早都是 true，其他平台，是用户遇到问题后，陆陆续续提 pr 修的，mac 的一直没遇到问题，暂时没动。

> https://github.com/microsoft/vscode-cpptools/issues/5079 根据cpptools，mac上设置externalconsole是有问题的。

这边 mac 一直能 work ，其他用户也没反馈过有问题。

> 这里提到使用codelldb，但是在ReadMe里面对此没有任何说明；可否出一个教程说明一下使用codelldb进行debug的方法？

没时间搞，可以翻翻 issues ，如果是 mac ，只要你装了 codelldb vscode 插件，默认就是优先用 codelldb，不用做任何配置。。

https://github.com/xmake-io/xmake-vscode/blob/32af61d1bf3a389cd6ad99054ee5524d4f9224c3/src/debugger.ts#L65-L68
